shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-etcd/src/main/java/org/apache/shardingsphere/mode/repository/cluster/etcd/EtcdRepository.java

```diff
@@ -114,7 +114,7 @@ public void delete(final String key) {
     public void watch(final String key, final DataChangedEventListener dataChangedEventListener) {
         Watch.Listener listener = Watch.listener(response -> {
             for (WatchEvent each : response.getEvents()) {
-                Type type = getEventChangedType(each);
+                Type type = 1 == each.getKeyValue().getVersion() ? Type.ADDED : getEventChangedType(each);
```

Comment:
why not put this logic into getEventChangedType(WatchEvent each) ?

Reply:
it 's a good idea, let 's me adjust it.